File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private List<Integer> res;
    public List<Integer> preorderTraversal(TreeNode root) {
        res=new ArrayList<>();
        preoreder(root);
        return res ;
        
    }
    private void preoreder(TreeNode n){
        if(n==null){
            return ;

        }
        res.add(n.val);
        preoreder(n.left);
        preoreder(n.right);
    }








}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(Deque, self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        q=deque([(root,root.val)])
        res=0
        while q :

            node , m = q.popleft()
            if node.val>=m :
                res+=1
            if node.left :
                q.append((node.left ,max(m,node.left.val)))
            if node.right :
                q.append((node.right,max(m,node.right.val)))
        return res







        